ddtrace/sampler.py

```diff
@@ -344,23 +344,16 @@ def sample(self, span):
         :returns: Whether the span was sampled or not
         :rtype: :obj:`bool`
         """
-        # If there are rules defined, then iterate through them and find one that wants to sample
-        sampler = None  # type: Optional[Union[SamplingRule, RateLimiter]]
-
         # Go through all rules and grab the first one that matched
         # DEV: This means rules should be ordered by the user from most specific to least specific
         for rule in self.rules:
             if rule.matches(span):
                 sampler = rule
                 break
         else:
-            # No rules matches so use agent based sampling
+            # No rules match so use agent based sampling
             return super(DatadogSampler, self).sample(span)
 
-        # DEV: This should never happen, but since the type is Optional we have to check
```

Comment:
This can never happen based on the for-else loop above, the above Optional type is unnecessary and so is this check